application/hubspot/v0/config/definition.json

```diff
@@ -0,0 +1,30 @@
+{
+  "availableTasks": [
+    "TASK_GET_CONTACT",
+    "TASK_CREATE_CONTACT",
+    "TASK_GET_DEAL",
+    "TASK_CREATE_DEAL",
```

Comment:
Question
It seems the design here is like
1. Get Contact
2. If there is no Contact, Create Contact (with company info)
3. Get Company
4. If there is no Company, Create Company
5. Get Deal
6. If there is no Deal, Create Deal
7. Get Ticket
8. If there is no Ticket, Create Ticket
9. Get Thread
10. If there is update from the conversation, Insert Message.

So, now, Does it seem we miss insert Message to Thread?

Reply:
Question

> As a sales manager, he/she wants to supervise the status of each client.

If you fake the sales tool component first, is your design able to fulfil this story request?

If my understanding is correct, we may miss company-id in Contact object

Reply:
> question
Do we need TASK_UPDATE_DEAL for this story?

> Question
It seems the design here is like

I will answer these questions together. 
1. First regarding the design. Initially when I was designing the component, I was thinking, how does HubSpot component fit together with the other already-made components and future possible components. For the "create" tasks, such as "create deal" and "create contact", I thought users will benefit from this when "Run on Event" pipeline is ready. Example: A WhatsApp message from a new client -> Create a new contact in HubSpot. For the "get" tasks, it is mainly for summarization purpose. With an email of the client, all the information about the client is obtainable (using the retrieve association task), then afterward can use LLM to summarize all those information.

2. Second, regarding the TASK_UPDATE_DEAL. Initially I thought update was not needed because I don't see the use case of it. I don't see how the output of text summarization will lead to the need of update. But after working with WhatsApp component, I can see how it can benefit from it.  

WhatsApp component has a [catalog template](https://developers.facebook.com/docs/whatsapp/business-management-api/message-templates/catalog-templates/) which allow clients to shop directly using WhatsApp. Once again, when the "Run on Event" pipeline is ready, this can automatically trigger the deal status to change, perhaps maybe from "qualified to buy" to "deal secured. But as of now, I am unsure if I should make the task now, feel free to give me an input.

> So, now, Does it seem we miss insert Message to Thread?

Yes, I was debating on this as well. I think I will create it after all.

> If you fake the sales tool component first, is your design able to fulfil this story request?

If my understanding is correct, we may miss company-id in Contact object

Yes, I think you are right after all. I will provide options for the user to be able to create contact -> object association (including company).

I am not sure what you mean by fake the sales tool, feel free to elaborate it thanks.

Reply:
@AmeliaCelline 
> Second, regarding the TASK_UPDATE_DEAL. I thought update was not needed because I don't see the use case of it. I don't see how the output of text summarization will lead to the need of update.

Maybe I missed some points. But, as I known, deal means the case that a sales is processing.
So, it means when a sales have a update with their clients in WhatsApp, the deal will be updated according to their message.

e.g. when a client said "I have sent the contract back to you", it means the deal should be updated as "contracting status".

Is this correct under this domain or industry?

> I am not sure what you mean by fake the sales tool, feel free to elaborate it thanks.

Because we do not have any Sales tool component like WhatsApp, the story must not be done by this PR.
So, fake it means "we assume we have WhatsApp, after we add HubSpot, could we fulfil this story?"

Reply:
> Maybe I missed some points. But, as I known, deal means the case that a sales is processing.
So, it means when a sales have a update with their clients in WhatsApp, the deal will be updated according to their message.

Yes, I think you are right. The update task does have its use case once more sales tool component is integrated. I think I can make "Update Ticket task" and "Update Deal task" as soon as I am done with WhatsApp "send message task".

Note: Just wanted to let you know that unfortunately, the WhatsApp component will not have "receive message" task, as it can only be implemented through webhook.

Reply:
@chuang8511 Maybe it is better if I create a new PR specifically for those 2 update tasks? I think this PR is getting too big

Reply:
Want to make sure if we lack these 2 tasks, can you build a demo pipeline like I mentioned in Slack?
If yes, I think you can do it in another PR.
If no, I think you need to do it now.

Reply:
> the WhatsApp component will not have "receive message" task, as it can only be implemented through webhook.

Does it mean there is no API to fetch users' message in a Chatroom?

Reply:
> Want to make sure if we lack these 2 tasks, can you build a demo pipeline like I mentioned in Slack?
If yes, I think you can do it in another PR.
If no, I think you need to do it with this PR.

Yes, I believe we can build a demo pipeline without those two tasks.

> Does it mean there is no API to fetch users' message in a Chatroom?
From my understanding so far, the official WhatsApp Business Platform API cannot retrieve user messages in a chatroom. The only way to get the content of the message is when we receive the message notification sent to the webhook endpoint that we set up.